bin/compile

```diff
@@ -93,6 +93,11 @@ fi
 
 speak "Running stack"
 cd "$WORK_DIR"
+# Makefile support (if you need other build tools than Stack that you need to use in conjunction)
+if [ -f $BUILD_DIR/Makefile ]; then
+    speak "Makefile detected"
+    make -C $BUILD_DIR
+fi
```

Comment:
>  I would like the option of executing the default Makefile action rather than the default Stack setup and build commands if a Makefile is provided.

This runs the Makefile and the stack commands - how about (notice the quotes around the variables, too):

```bash
if [ -f "$BUILD_DIR/Makefile" ]; then
    speak "Makefile detected"
   make -C "$BUILD_DIR"
else
    stack setup
    stack build --fast --copy-bins
fi
```